tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -373,6 +362,31 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     return OkStatus();
   }
 
+  Status ExchangeBinaryOpAndSlice(HloInstruction *instr, HloInstruction *slice,
```

Comment:
This function doesn't really *exchange* the binary op and the slice.  It moves the slice up or down -- I'm not sure, but I think downward?  In which case a better name might be SinkSliceBelowBinaryOp or something, to indicate that this only goes in one direction.

Except also this function is not generic to any binary op.  One of the two operands of the binary op must be a broadcast.  SinkSliceBelowBinaryOpWithBroadcastedOperand?  It's a mouthful but it does get the point across.

Reply:
SinkSliceBelowBinaryOp could be a good compromise.